### PR TITLE
Generate fluent config for new syslog plugin

### PR DESCRIPTION
**Pull request by vimalk78**
Old Plugin can be selected by adding annotation `clusterlogging.openshift.io/useoldremotesyslogplugin` to LogForwarding CR
Added e2e tests for new syslog plugin with/without TLS
Added unit tests

fixes: https://issues.redhat.com/browse/LOG-656

-- Pending
Test cases for RFC 3164 & 5424
generate syslog config with tag_key,use_record,and remove_tag_prefix

### CODE REVIEW

**Comment by vimalk78** [comment]
> Looks good to me, comments are minor and can be addressed when we move this to the new API.
> Where's the ruby code?
> /lgtm

Ruby plugin was added in this PR
https://github.com/openshift/origin-aggregated-logging/pull/1860

File: pkg/generators/forwarding/factory.go [hunk 1]
```diff
@@ -9,10 +9,10 @@ import (
 )
 
 //NewConfigGenerator create a config generator for a given collector type
-func NewConfigGenerator(collector logging.LogCollectionType, includeLegacyForwardConfig, includeLegacySyslogConfig bool) (ConfigGenerator, error) {
+func NewConfigGenerator(collector logging.LogCollectionType, includeLegacyForwardConfig, includeLegacySyslogConfig, useOldRemoteSyslogPlugin bool) (ConfigGenerator, error) {
 	switch collector {
 	case logging.LogCollectionTypeFluentd:
-		return fluentd.NewConfigGenerator(includeLegacyForwardConfig, includeLegacySyslogConfig)
+		return fluentd.NewConfigGenerator(includeLegacyForwardConfig, includeLegacySyslogConfig, useOldRemoteSyslogPlugin)
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply pkg/generators/forwarding/factory.go:15]: We decided in the sprint planing meeting to keep the old plugin.

File: pkg/apis/logging/v1alpha1/forwarding_types.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ const (
 	//OutputTypeForward configures the pipeline to send messages via Fluent's secure forward
 	OutputTypeForward OutputType = "forward"
 
-	//OutputTypeSyslog configures pipeline to send messages to an external syslog server through docebo/fluent-plugin-remote-syslog
+	//OutputTypeSyslog configures pipeline to send messages to an external syslog server through docebo/fluent-plugin-remote-syslog or dlackty/fluent-plugin-remote_syslog
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply pkg/apis/logging/v1alpha1/forwarding_types.go]: updated.

**Comment by alanconway** [comment]
/lgtm
Will review the exposed configuration again when this is migrated to the GA log forwarder API.

**Pull request merged**